x/compute/internal/keeper/bench_test.go

```diff
@@ -304,4 +341,129 @@ func TestRunBenchmarks(t *testing.T) {
 		timer, _ := timers[name]
 		timer.PrintReport()
 	}
+
+	// Run the query benchmarks.
+	// Notice that it's done in the same test, as the result of the queries depend on the
+	// transactions above.
+	RunQueryBenchmarks(t, keeper, ctx, contractAddr, creator, creatorPriv, viewingKey)
+}
+
+func RunQueryBenchmarks(
+	t *testing.T, keeper Keeper, ctx sdk.Context,
+	contractAddr sdk.AccAddress, creator sdk.AccAddress, creatorPriv crypto.PrivKey,
+	viewingKey string,
+) {
+	timers := make(map[string]BenchTime)
+
+	queryCases := map[string]struct {
+		bench            Bench
+		queryFunctionPtr func(
+			t *testing.T, keeper Keeper, ctx sdk.Context,
+			contractAddr sdk.AccAddress, creator sdk.AccAddress, creatorPriv crypto.PrivKey,
+			viewingKey string,
+		)
+		loops uint64
+	}{
+		"Query with viewing-key": {
+			bench:            WithPermit,
+			queryFunctionPtr: vkBenchRun,
+			loops:            10,
+		},
+		"Query with permit": {
+			bench:            Balance,
+			queryFunctionPtr: PermitBenchRun,
+			loops:            10,
+		},
+	}
+
+	for name, tc := range queryCases {
+		t.Run(name, func(t *testing.T) {
+			timer := NewBenchTimer(name, tc.bench)
+			for i := uint64(1); i < tc.loops+1; i++ {
+				start := time.Now()
+				// call bench
+				tc.queryFunctionPtr(t, keeper, ctx, contractAddr, creator, creatorPriv, viewingKey)
+				elapsed := time.Since(start)
+				timer.AppendResult(elapsed, 0)
+			}
+			timers[name] = timer
+		})
+	}
+
+	for name := range queryCases {
+		timer, _ := timers[name]
+		timer.PrintReport()
+	}
+}
+
+func vkBenchRun(
+	t *testing.T, keeper Keeper, ctx sdk.Context,
+	contractAddr sdk.AccAddress, creator sdk.AccAddress, _ crypto.PrivKey,
+	viewingKey string,
+) {
+	vkQueryMsg := fmt.Sprintf(`{"balance":{"address": "%s", "key":"%s"}}`, creator, viewingKey)
+
+	expectedResult := `{"balance":{"amount":"42"}}`
+	queryAndVerify(t, keeper, ctx, contractAddr, vkQueryMsg, expectedResult)
+}
+
+func PermitBenchRun(
+	t *testing.T, keeper Keeper, ctx sdk.Context,
+	contractAddr sdk.AccAddress, creator sdk.AccAddress, creatorPriv crypto.PrivKey,
+	_ string,
+) {
+	permitSignature := generatePermitSignature(ctx, keeper, contractAddr, creator, creatorPriv)
```

Comment:
The changes make things a little better, but you're still benchmarking both createPermitQueryMsg which signs the permit and the query itself - I have a feeling that the 1-2 ms difference between permits and viewing keys might be because of it - signing something with a private key tends to be non-negligible when benchmarking